src/connector/src/sink/elasticsearch.rs

```diff
@@ -109,23 +136,39 @@ impl EsStreamChunkConverter {
             <Utf8ArrayBuilder as risingwave_common::array::ArrayBuilder>::new(chunk.capacity());
         let mut json_builder =
             <JsonbArrayBuilder as risingwave_common::array::ArrayBuilder>::new(chunk.capacity());
+        let mut index_builder =
+            <Utf8ArrayBuilder as risingwave_common::array::ArrayBuilder>::new(chunk.capacity());
         for (op, row) in chunk.rows() {
             ops.push(op);
-            let json = JsonbVal::from(Value::Object(self.json_encoder.encode(row)?));
             risingwave_common::array::ArrayBuilder::append(
                 &mut id_string_builder,
                 Some(&self.build_id(row)?),
             );
+            if let Some(index) = self.index_column {
+                risingwave_common::array::ArrayBuilder::append(
+                    &mut index_builder,
+                    Some(
+                        row.datum_at(index)
+                            .ok_or_else(|| anyhow!("No value find in row, index is {}", index))?
+                            .into_utf8(),
+                    ),
+                );
+            } else {
+                risingwave_common::array::ArrayBuilder::append(&mut index_builder, None);
```

Comment:
nits: `index_builder.append_null()`

btw, I think all `risingwave_common::array::ArrayBuilder::append(&mut builder, ...)` can be simplified to `builder.append(...)` in `convert_chunk`.